exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV2ExporterHandler.java

```diff
@@ -0,0 +1,249 @@
+  private static Attributes toAttributesProto(
+      io.opencensus.trace.export.SpanData.Attributes attributes) {
+    Attributes.Builder attributesBuilder =
+        toAttributesBuilderProto(
+            attributes.getAttributeMap(), attributes.getDroppedAttributesCount());
+    attributesBuilder.putAttributeMap(AGENT_LABEL_KEY, AGENT_LABEL_VALUE);
+    return attributesBuilder.build();
+  }
+
+  private static Attributes.Builder toAttributesBuilderProto(
+      Map<String, io.opencensus.trace.AttributeValue> attributes, int droppedAttributesCount) {
+    Attributes.Builder attributesBuilder =
+        Attributes.newBuilder().setDroppedAttributesCount(droppedAttributesCount);
+    for (Map.Entry<String, io.opencensus.trace.AttributeValue> label : attributes.entrySet()) {
+      attributesBuilder.putAttributeMap(label.getKey(), toAttributeValueProto(label.getValue()));
+    }
+    return attributesBuilder;
+  }
+
+  private static Status toStatusProto(io.opencensus.trace.Status status) {
+    Status.Builder statusBuilder = Status.newBuilder().setCode(status.getCanonicalCode().value());
+    if (status.getDescription() != null) {
+      statusBuilder.setMessage(status.getDescription());
+    }
+    return statusBuilder.build();
+  }
+
+  private static TruncatableString toTruncatableStringProto(String string) {
+    return TruncatableString.newBuilder().setValue(string).setTruncatedByteCount(0).build();
+  }
+
+  private static com.google.protobuf.Timestamp toTimestampProto(Timestamp timestamp) {
+    return com.google.protobuf.Timestamp.newBuilder()
+        .setSeconds(timestamp.getSeconds())
+        .setNanos(timestamp.getNanos())
+        .build();
+  }
+
+  private static AttributeValue toAttributeValueProto(
+      io.opencensus.trace.AttributeValue attributeValue) {
+    final AttributeValue.Builder attributeValueBuilder = AttributeValue.newBuilder();
+    attributeValue.match(
+        new Function<String, Void>() {
+          @Override
+          public Void apply(String stringValue) {
+            attributeValueBuilder.setStringValue(toTruncatableStringProto(stringValue));
+            return null;
+          }
+        },
+        new Function<Boolean, Void>() {
+          @Override
+          public Void apply(Boolean booleanValue) {
+            attributeValueBuilder.setBoolValue(booleanValue);
+            return null;
+          }
+        },
+        new Function<Long, Void>() {
+          @Override
+          public Void apply(Long longValue) {
+            attributeValueBuilder.setIntValue(longValue);
+            return null;
+          }
+        },
+        Functions.<Void>returnNull());
+    return attributeValueBuilder.build();
+  }
+
+  @Override
+  public void export(Collection<SpanData> spanDataList) {
+    List<Span> spans = new ArrayList<>(spanDataList.size());
+    for (SpanData spanData : spanDataList) {
+      spans.add(generateSpan(spanData));
+    }
+    traceServiceClient.batchWriteSpans(projectName, spans);
```

Comment:
maybe add a comment what this implies? Ex is this async

Reply:
done.